### DISCUSSION TITLE
Amasty Shopby filters not working

### DISCUSSION THREAD

**Discussion by rutulritz** [breezefront/community#29] [3. Integrations]
Hi there,

I installed the Breeze Evolution theme on my Magento website. Also I have installed Amasty shopby module in the website.

Now the FIlters showing on the Left navigation but the Ajax call and other jquery functions not working(Ex:Price Slider filter, SHow/Hide More options, etc.).

Can you please guide me to fix the issue for the same.

Thanks.
Ritz

**Comment by vovayatsyuk** [comment]
We do not have an integration for Amasty Shopby module yet 🙁

P.s. If you are interested, we have integration with our own [AjaxLayeredNavigation](https://swissuplabs.com/layered-navigation-extension-for-magento-2.html)
The integration is not full yet, but the main features are already working.

**Comment by rutulritz** [comment]
HI @vovayatsyuk ,

Thanks for the quick response. 

Can you please guide me on the same to skip particular JS from breeze-evolution? Or any overriden need to be changed for the adding custom scripts?

Because the whole Magento require JS removed from website.

Thanks

**Reply by vovayatsyuk** [reply]
We have a small guide on how to add custom js: https://breezefront.com/custom-javascript

**Reply by rutulritz** [reply]
But some third party JS already there in module which I want to override in breeze theme

**Reply by vovayatsyuk** [reply]
Can you please describe what are you trying to do?

**Reply by rutulritz** [reply]
I installed Amasty Shopby extension. but due to Breeze theme its Price slider ui and ajax is not working. SO How can I ovverride the module or its JS files ?

**Reply by vovayatsyuk** [reply]
Take a look at Amasty_Xsearch integration: https://github.com/breezefront/module-breeze-amasty-xsearch

1. Breeze components are registered via layout update file: https://github.com/breezefront/module-breeze-amasty-xsearch/blob/master/view/frontend/layout/breeze_default.xml
2. Overriden files: https://github.com/breezefront/module-breeze-amasty-xsearch/tree/master/view/frontend/web/js

**Reply by rutulritz** [reply]
Thanks for the response.
I am doing the same by referring the given modules.
All th JS component added in the XML from the map : *  of require-config.js of module. in breeze_default.xml

Now I am stuck to include the JS part given below. How can I add it in xml? can you give me an example?
```
config: {
        mixins: {
            'jquery/ui-modules/widgets/slider': {
                'Amasty_Shopby/js/mixins/slider': true
            }
        }
    },
    paths: {
        'swiper': 'Amasty_ShopbyBase/js/swiper.min'
    },
    shim: {
        'swiper': {
            deps: [ 'jquery' ]
        }
    }

```

Thanks

![image](https://user-images.githubusercontent.com/9527439/203022497-664bdf22-39d4-4413-9c12-91ad9f3caab3.png)

**Reply by vovayatsyuk** [reply]
Please do not copy-paste everything from luma to breeze. Such approach will just confuse you.

Start from one component only. When it will be added to the page and will work then proceed with the next one.

p.s. In your specific case with shims, mixins, etc, you don't need them in breeze because we do not have jquery/requirejs.

**Reply by rutulritz** [reply]
Ok, So I am trying to add fancybox or slick slider js component in breeze_default.xml as per refernce module. It is added on page but giving error as below . How can I override the component of this like of JS on my page?
```

<?xml version="1.0"?>
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <body>
        <referenceBlock name="breeze.js">
            <arguments>
                <argument name="bundles" xsi:type="array">
                    <item name="product" xsi:type="array">
                        <item name="items" xsi:type="array">
                            
                            <item name="Smart_SizeChart/js/jquery.fancybox.pack" xsi:type="array">
                                <item name="path" xsi:type="string">Smart_SizeChart/js/jquery.fancybox.pack</item>
                            </item>
                            
                        </item>
                    </item>
                </argument>
            </arguments>
        </referenceBlock>
    </body>
</page>

```

![image](https://user-images.githubusercontent.com/9527439/203308674-6720ec67-218f-4e57-a1a9-074baf74236f.png)

**Reply by vovayatsyuk** [reply]
We don't have jQuery in Breeze. Breezefront is not a regular Magento theme. It uses a completely new js stack without jQuery.

If you want to save the speed of BreezeBlank/Evolution, please do not add unnecessary libraries.

Here is a code that shows a popup without any additional js libraries:

```html
<button class="button primary modal-opener">Show popup</button>

<div style="display:none" data-mage-init='{
        "Magento_Ui/js/modal/modal": {
            "trigger": ".modal-opener"
        }
    }'>
    Modal content
</div>
```

Just add this code to some of your templates, or cms page.

**Reply by rutulritz** [reply]
Thanks for the reply @vovayatsyuk , above code is  helpful for me.

Can you help to create price slider using breeze js stack ?
![image](https://user-images.githubusercontent.com/9527439/203789172-67c93355-c7b4-465c-b1e9-5f40265809d7.png)


It is required for the price filtering in our website.

Thanks in advance.

**Reply by vovayatsyuk** [reply]
In our AjaxLayeredNavigation, we use this one: https://github.com/swissup/range-slider

**Comment by rutulritz** [comment]
Hi 

I am getting error as below when validating the form in magento 2 
Uncaught TypeError: $(...).mage is not a function

Can you please help me to fix this error on the breeze revolution theme?

thanks in advance

**Reply by vovayatsyuk** [reply]
$.fn.mage is just a proxy to init other components/widgets, so integration of this code depends on arguments of the mage function.

**Comment by rutulritz** [comment]
Hi @vovayatsyuk ,

How Can I disable breeze experience for particular urls like below :

- customer/account/login/
- customer/account/create/
- customer/account/create/phone/999999/email/teste@email.com

For last url it will be dynamic with phone & email params 

Can you please guide me on this? 

Thanks

**Reply by dimdimich** [reply]
Please simply check https://breezefront.com/settings. You have "Disable Breeze for specified URLs" config option where you can simply put this urls.

**Reply by rutulritz** [reply]
@dimdimich I already checked this. But my concern for the third url which will be dynamic. [customer/account/create/phone/999999/email/[teste@email.com](mailto:teste@email.com)]

**Reply by vovayatsyuk** [reply]
Omit the dynamic parameters. Use “customer/account/create” url.

**Comment by rutulritz** [comment]
Hi @vovayatsyuk ,

Can you please help me to override my below component using breeze_default.xml.

I have added below code in MagentoRoot/app/design/frontend/theme/default/Amasty_Shopby/layout/breeze_default.xml 
<item name="amShopbyFilterSlider" xsi:type="array">
    <item name="path" xsi:type="string">Amasty_Shopby/js/components/filterslider</item>
</item>

I created new file with under MagentoRoot/app/design/frontend/theme/default/Amasty_Shopby/web/js/components/filterslider.js

added below code in js file:
```

define([
    'jquery',
    'amShopbyFilterAbstract',
    'amshopby_color',
    'amShopbyHelpers',
    'jquery-ui-modules/slider',
    'mage/tooltip',
    'amShopbyFiltersSync'
], function ($, amShopbyFilterAbstract, colorHelper, helpers) {
    // (function () {
    'use strict';

    $.widget('filterslider', {
        component: 'amShopbyFilterSlider',
        options: {
            gradients: {}
        },
        selectors: {
            value: '[data-amshopby-slider-id="value"]',
            range: '.ui-slider-range',
            slider: '[data-amshopby-slider-id="slider"]',
            display: '[data-amshopby-slider-id="display"]',
            container: '[data-am-js="slider-container"]',
            tooltip: '[data-amshopby-js="slider-tooltip"]',
            corner: '[data-amshopby-js="slider-corner"]',
            handle: '.ui-slider-handle',
            topNav: '.amasty-catalog-topnav'
        },
        classes: {
            tooltip: 'amshopby-slider-tooltip',
            corner: 'amshopby-slider-corner',
            styleDefault: '-default',
            loaded: '-loaded'
        },
        attributes: {
            tooltip: 'slider-tooltip',
            corner: 'slider-corner'
        },
        slider: null,
        value: null,
        display: null,

        /**
         * @inheritDoc
         */
        create: function () {
            console.log('slider init');
            helpers.jqueryWidgetCompatibility(
                'jquery/ui-modules/widgets/slider',
                'slider',
                function () {
                    this._initializeWidget();
                }.bind(this)
            );
        },
   .............
.......
....
});

})();
```

but getting the below errors : 
1. Uncaught TypeError: define(...) is not a function.
2. Uncaught TypeError: helpers is undefined

Please let me know where I am making mistake to rewrite the JS file under breeze evolution.

Please let me know if you have any questions.

Thanks

**Reply by vovayatsyuk** [reply]
1.  Most likely you are using the old breeze module.
2. See how we manage the similar case with amasty helpers dependency: 
    1. https://github.com/breezefront/module-breeze-amasty-xsearch/blob/master/view/frontend/layout/breeze_default.xml#L12-L14
    2. https://github.com/breezefront/module-breeze-amasty-xsearch/blob/master/view/frontend/web/js/helpers.js#L24

**Reply by rutulritz** [reply]
Hi @vovayatsyuk 

I have below helprs.js exist for Amasty module . 

```
define([
    'jquery'
], function ($) {
    'use strict';

    /**
     * Cut zeroes from value
     *
     * @param {Number} digits - signs after dot
     * @returns {String | Number}
     */
    // eslint-disable-next-line no-extend-native
    Number.prototype.amToFixed = function (digits, hideDigitsAfterDot) {
        var value = this.toFixed(digits);

        if (hideDigitsAfterDot && parseInt(value) == this) {
            value = parseInt(value)
        }

        return value;
    };

    return {
        /**
         * Compatibility with 2.4.4
         * Store jquery-ui-widget to $.ui if it not exist
         *
         * @public
         * @param {String} path - direct path to jquery widget file
         * @param {String} widget - widget name
         * @param {Function} callback
         * @returns {void}
         */
        jqueryWidgetCompatibility: function (path, widget, callback) {
            if ($.ui[widget]) {
                callback();

                return;
            }

            // eslint-disable-next-line global-require
            require([path], function (instance) {
                $.ui[widget] = instance;

                callback();
            });
        }
    };
});

```

So how can we update it ?

Thanks

**Reply by rutulritz** [reply]
I added like this in breeze_default.xml 

```
<item name="amShopbyFilterSlider" xsi:type="array">
                                <item name="path" xsi:type="string">Amasty_Shopby/js/components/filterslider</item>
                                <item name="import" xsi:type="array">
                                    <item name="helpers" xsi:type="string">Amasty_Shopby/js/utils/helpers</item>
                                </item>
                            </item>
```

**Reply by vovayatsyuk** [reply]
This `helpers.jqueryWidgetCompatibility` function is useless in breeze. If I'd work on the integration of this module I would simply throw it away.

p.s. Please do not copy-paste all of the Luma-based code into breeze integration. If some module is packed with tons of scripts, you should make a decision to remove most of the js code in favor of simplicity and site speed when integrating it.

**Comment by rutulritz** [comment]
Hi @vovayatsyuk ,

How can I disable breeze experience for particular elements [div, selectors, any html section]?

I need to disable it for some sections like in my case for Megamenu.

Please guide me on the same

Thanks

**Reply by vovayatsyuk** [reply]
It's not possible.

**Reply by rutulritz** [reply]
So How can I disable minicart loading via breeze ? I just want default magento minicart on every pages

**Reply by vovayatsyuk** [reply]
It's not possible. You can't use luma scripts partially. Breeze allows you to disable it completely on specific pages only.

**Reply by rutulritz** [reply]
Oh, Ok. Thanks

How Can I write javascript on phtml files to do below actions

```
require(['jquery'],function($){
      $(document).ready(function() {
           $('.classs').on('click', function(){
                 .....some actions code
           })
     });
});
```

**Reply by vovayatsyuk** [reply]
https://breezefront.com/integration-tips#migrate-inline-scripts

**Reply by rutulritz** [reply]
Hi 

for my megmenu there are too many inline scripts in one phtml. 

Can you guide me by giving one example by fixing in my code so I can refer the same ?

It will be very helpful as I am not a full stack or frontend developer.

[topmenu-phtml.zip](https://github.com/breezefront/community/files/10228678/topmenu-phtml.zip)


THanks.

**Reply by vovayatsyuk** [reply]
Sorry, but I can't. This file has about 400 lines of mixed js and php code. It's a mess. 

I recommend you to use our Navigationpro module instead.
